packages/jwt-middleware/index.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+const Assert = require('assert');
+const ExpressJwt = require('express-jwt');
+const jsonwebtoken = require('jsonwebtoken');
+const JwksRsa = require('jwks-rsa');
+
+const WT_AUTH_EXEC = 'wt-authorize-execution';
+const WT_META_EXEC_SCOPE ='wt-execution-scope';
+const WT_META_ADMIN_SCOPE = 'wt:admin';
+const WT_META_ISS = 'wt-execution-iss';
+const WT_META_AUD ='wt-execution-aud';
+
+const AUTHZ = 'authorization';
+const BEARER = /^bearer\s+(.+)\s*$/i;
+
+module.exports = function middlewareFactory(){
+    let cachedMiddlewareInstance;
+
+    return function(req, res, next) {
+        if (!cachedMiddlewareInstance) {
+            cachedMiddlewareInstance = createMiddleware(req);
+        }
+
+        return cachedMiddlewareInstance(req, res, next);
+    }
+}
+
+function createMiddleware(req) {
+
+    const iss = req.webtaskContext.meta[WT_META_ISS];
+    const aud = req.webtaskContext.meta[WT_META_AUD]
+
+    // setup JWT middleware
+
+    const loadRsaKey = JwksRsa.expressJwtSecret({
+        cache: true,
+        rateLimit: true,
+        jwksRequestsPerMinute: 5,
+        jwksUri: `${iss}`
+    });
+
+    const middleware = ExpressJwt({
+        algorithms: ['RS256'],
+        audience: aud,
+        issuer: iss,
+        secret: loadRsaKey
+    });
+
+    return function(req, res, next) {
+
+        const ctx = req.webtaskContext;
+
+        // if wt-authorize-execution is set to true (non-zero) then proceed with authn and authz
+
+        if (ctx.meta[WT_AUTH_EXEC] && ctx.meta[WT_AUTH_EXEC] !== "0"){
+
+            if (req.headers[AUTHZ]) {
+
+                // authentication
+
+                try {
+
+                    return middleware(req, res, error => {
+
+                        if (error) {
+
+                            let e = new Error(error.name);
+                            e.statusCode = error.status;
+                            return next(e);
+
+                        } else {
+
+                            // authorization
+
+                            const match = (req.headers[AUTHZ] || '').trim().match(BEARER);
+
+                            let jwt;
+
+                            try {
+                                jwt = jsonwebtoken.decode(match[1]);
+                                if (!jwt) throw new Error();
+                            } catch(e){
+                                const error = new Error('UnauthorizedError');
+                                error.statusCode = 401;
+                                return next(error);
+                            }
+
+                            // verify claims
+                            const scope = jwt.scope.split(' ');
+
+                            if (
+                                scope.indexOf(WT_META_ADMIN_SCOPE) === -1 
```

Comment:
`scope` is now an array of scopes and you'll need to iterate over this array... correct?

Reply:
You shouldn't have to since `indexOf` will look through the array to see if `wt:admin` is present.  It will also do that for `wt:owner:{container}` and then for the value from `wt_execution_scope`.  If all three returned `-1` then no valid scope is present in the meta`.

Reply:
`[].indexOf(undefined)` will always return `-1` unless the primitive `undefined` somehow ended up in the scope array, but I can agree that better safe than sorry. Fixed and checked in.